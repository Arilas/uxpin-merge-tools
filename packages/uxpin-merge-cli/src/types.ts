export { ComponentImplementationInfo } from './steps/discovery/component/ComponentInfo';
export { ImplSerializationResult } from './steps/serialization/component/implementation/ImplSerializationResult';
export { serializeJSComponent } from './steps/serialization/component/implementation/javascript/serializeJSComponent';
export { serializeTSComponent } from './steps/serialization/component/implementation/typescript/serializeTSComponent';
export { DesignSystemSnapshot } from './steps/serialization/DesignSystemSnapshot';
