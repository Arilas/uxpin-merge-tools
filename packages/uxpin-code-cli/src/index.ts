export { getDesignSystemMetadata } from './getDesignSystemMetadata';
export { getDesignSystemSummary } from './getDesignSystemSummary';
