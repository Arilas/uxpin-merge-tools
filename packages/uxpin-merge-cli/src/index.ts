import 'source-map-support/register';

export { runProgram } from './program/runProgram';
export { Command } from './program/command/Command';
export { getToolVersion } from './program/utils/version/getToolVersion';

export { ComponentImplementationInfo } from './steps/discovery/component/ComponentInfo';
export { ImplSerializationResult } from './steps/serialization/component/implementation/ImplSerializationResult';
export { serializeJSComponent } from './steps/serialization/component/implementation/javascript/serializeJSComponent';
export { serializeTSComponent } from './steps/serialization/component/implementation/typescript/serializeTSComponent';
export { DesignSystemSnapshot } from './steps/serialization/DesignSystemSnapshot';
