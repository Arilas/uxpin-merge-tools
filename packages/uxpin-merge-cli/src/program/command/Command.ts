export enum Command {
  DUMP = 'dump',
  EXPERIMENT = 'experiment',
  INIT = 'init',
  CREATE_APP = 'create-app',
  GENERATE_PRESETS = 'generate-presets',
  PUSH = 'push',
  SERVER = 'server',
  SUMMARY = 'summary',
}

export const DEFAULT_COMMAND:Command = Command.EXPERIMENT;
