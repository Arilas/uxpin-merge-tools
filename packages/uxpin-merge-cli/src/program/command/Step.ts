import { DSMetadata } from '../DSMeta';

export interface Step {
  shouldRun:boolean;
  exec:(infos:DSMetadata) => any;
}
