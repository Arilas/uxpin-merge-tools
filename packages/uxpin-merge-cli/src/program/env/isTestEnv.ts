import { Environment } from './Environment';

export function isTestEnv():boolean {
  return process.env.NODE_ENV === Environment.TEST;
}
