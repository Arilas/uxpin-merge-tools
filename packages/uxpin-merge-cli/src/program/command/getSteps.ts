import { BuildOptions } from '../../steps/building/BuildOptions';
import { ProgramArgs, PushProgramArgs, ServerProgramArgs } from '../ProgramArgs';
import { getDumpCommandSteps } from './dump/getDumpCommandSteps';
import { getPushCommandSteps } from './push/getPushCommandSteps';
import { getServerCommandSteps } from './server/getServerCommandSteps';
import { Step } from './Step';
import { getSummaryCommandSteps } from './summary/getSummaryCommandSteps';

export function getSteps(args:ProgramArgs):Step[] {
  if (args.command === 'server') {
    return getServerCommandSteps(getBuildOptions(args), args);
  }

  if (args.command === 'push') {
    return getPushCommandSteps(getBuildOptions(args));
  }

  if (args.command === 'dump') {
    return getDumpCommandSteps();
  }

  if (args.command === 'summary') {
    return getSummaryCommandSteps();
  }

  return [];
}

function getBuildOptions(args:ServerProgramArgs | PushProgramArgs):BuildOptions {
  const { cwd, webpackConfig, wrapper } = args;
  return {
    projectRoot: cwd,
    webpackConfigPath: webpackConfig,
    wrapperPath: wrapper,
  };
}
