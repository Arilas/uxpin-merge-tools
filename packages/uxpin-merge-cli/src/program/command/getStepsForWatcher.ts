import { ProgramArgs } from '../args/ProgramArgs';
import { getExperimentationWatchCommandSteps } from './experimentation/getExperimentationWatchCommandSteps';
import { Step } from './Step';

export function getStepsForWatcher(args:ProgramArgs):Step[] {
  if (args.command === 'experiment') {
    return getExperimentationWatchCommandSteps(args);
  }

  return [];
}
