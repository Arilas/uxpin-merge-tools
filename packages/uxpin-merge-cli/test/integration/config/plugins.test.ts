import { resolve } from 'path';
import { DirectoryResult } from 'tmp-promise';
import { Command } from '../../../src';
import { Environment } from '../../../src/program/env/Environment';
import { runUXPinMergeCommand } from '../../utils/command/runUXPinMergeCommand';
import { setTimeoutBeforeAll } from '../../utils/command/setTimeoutBeforeAll';
import { setupTempProject } from '../../utils/temp/setupTempProject';

const CURRENT_TIMEOUT:number = 60000;

setTimeoutBeforeAll(CURRENT_TIMEOUT);

describe('Plugins support', () => {
  const sourceDir:string = resolve(__dirname, '../../resources/designSystems/twoComponentsWithConfig');
  const projectPath:string = resolve(__dirname, '../../../');
  const { getDirectory } = setupTempProject({
    gitOptions: { initialise: true },
    linkPackage: true,
    projectPath,
    sourceDir,
  });

  it('it should be possible to serialize component using custom plugin', async () => {
    // having
    const dir:DirectoryResult = getDirectory();

    // when
    const result:string = await runUXPinMergeCommand({
      cwd: dir.path,
      env: {
        UXPIN_ENV: Environment.TEST,
      },
      params: [
        Command.DUMP,
        '--config "./uxpin.plugin.config.js"',
        '--webpack-config "./webpack.config.js"',
      ],
    });

    // then
    expect(JSON.parse(result)).toMatchSnapshot({
      vcs: expect.objectContaining({
        branchName: expect.any(String),
        commitHash: expect.stringMatching(/[a-z0-9]+/),
      }),
    });
  });
});
