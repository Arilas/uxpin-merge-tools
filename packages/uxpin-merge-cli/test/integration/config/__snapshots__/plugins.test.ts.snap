// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plugins support it should be possible to serialize component using custom plugin 1`] = `
Object {
  "categorizedComponents": Array [
    Object {
      "components": Array [
        Object {
          "documentation": Object {
            "examples": Array [],
          },
          "info": Object {
            "dirPath": "src/components/Avatar",
            "implementation": Object {
              "framework": "reactjs",
              "lang": "typescript",
              "path": "src/components/Avatar/Avatar.tsx",
            },
            "presets": Array [
              Object {
                "path": "src/components/Avatar/presets/0-default.jsx",
              },
            ],
          },
          "name": "Avatar",
          "presets": Array [
            Object {
              "elements": Object {
                "1": Object {
                  "name": "Avatar",
                  "props": Object {
                    "imageUrl": "https://picsum.photos/100/100?image=1027",
                    "size": "small",
                  },
                },
              },
              "name": "default",
              "rootId": "1",
            },
          ],
          "properties": Array [
            Object {
              "description": "",
              "isRequired": false,
              "name": "nerdlihc",
              "type": Object {
                "name": "node",
                "structure": Object {},
              },
            },
            Object {
              "description": "",
              "isRequired": true,
              "name": "ezis",
              "type": Object {
                "name": "string",
                "structure": Object {},
              },
            },
            Object {
              "description": "",
              "isRequired": true,
              "name": "lrUegami",
              "type": Object {
                "name": "string",
                "structure": Object {},
              },
            },
          ],
        },
      ],
      "name": "General",
    },
    Object {
      "components": Array [
        Object {
          "documentation": Object {
            "examples": Array [],
          },
          "info": Object {
            "dirPath": "src/components/Button",
            "implementation": Object {
              "framework": "reactjs",
              "lang": "typescript",
              "path": "src/components/Button/Button.tsx",
            },
            "presets": Array [
              Object {
                "path": "src/components/Button/presets/0-default.jsx",
              },
            ],
          },
          "name": "Button",
          "presets": Array [
            Object {
              "elements": Object {
                "1": Object {
                  "name": "Button",
                  "props": Object {
                    "children": "Click me",
                    "isPrimary": true,
                  },
                },
              },
              "name": "default",
              "rootId": "1",
            },
          ],
          "properties": Array [
            Object {
              "description": "",
              "isRequired": true,
              "name": "nerdlihc",
              "type": Object {
                "name": "node",
                "structure": Object {},
              },
            },
            Object {
              "description": "",
              "isRequired": false,
              "name": "yramirPsi",
              "type": Object {
                "name": "boolean",
                "structure": Object {},
              },
            },
          ],
        },
      ],
      "name": "Forms",
    },
  ],
  "name": "Example Design System",
  "vcs": ObjectContaining {
    "branchName": Any<String>,
    "commitHash": StringMatching /\\[a-z0-9\\]\\+/,
  },
}
`;
